radixdlt/src/main/java/com/radixdlt/consensus/TimestampedVoteData.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.consensus;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.primitives.Longs;
+import com.radixdlt.serialization.DsonOutput;
+import com.radixdlt.serialization.DsonOutput.Output;
+import com.radixdlt.serialization.SerializerConstants;
+import com.radixdlt.serialization.SerializerDummy;
+import com.radixdlt.serialization.SerializerId2;
+import java.util.Objects;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * {@link VoteData} with a timestamp.
+ */
+@Immutable
+@SerializerId2("consensus.timestamped_vote_data")
+public final class TimestampedVoteData {
```

Comment:
Then each VoteData will (probably) be different for each node, and we will need to keep them all to validate the signatures.  This would greatly increase the quantity of data in QCs and Votes etc, particularly for circa 100 nodes.

Reply:
I'm not sure if I follow this properly. Doesn't the timestamp also need to be moved through the three chain in order to have consensus on it? Or am I missing how that's implemented?

Reply:
right but there's an issue with consensus in that everyone won't be able to agree on the same median timestamp unless it goes through the three phases